### ISSUE TITLE
doxygen output is broken

### ISSUE THREAD

**Issue by streichler** [StanfordLegion/legion#612]
For an example: https://legion.stanford.edu/doxygen/class_legion_1_1_physical_region.html
Right now it's worse than not having it at all, so we should either fix it or not have it at all.

**Comment by lightsighter** [comment]
How is it broken? It looks fine to me.

**Comment by lightsighter** [comment]
Note the doxygen output isn't broken as all the methods are there, it just sucks at rendering it with some of our deprecation macros. I'm inclined to label this a doxygen problem as it should be smart enough to ignore macros like this, or at least preprocess them to get the actual c++ code.

**Comment by streichler** [comment]
Hmm...  It looks like we can put something like this in the doxygen settings to have it eat that macro:
```
ENABLE_PREPROCESSING   = YES
MACRO_EXPANSION        = YES
EXPAND_ONLY_PREDEF     = YES
PREDEFINED             = LEGION_DEPRECATED(x)=
```
@elliottslaughter can we give that a try?  It also looks like doxygen has its own `\deprecated` tag, but it's not obvious to me how you'd get the LEGION_DEPRECATED macro to turn into that properly.

**Comment by elliottslaughter** [comment]
Pushed. Please check the output.

**Comment by streichler** [comment]
Looks much better.  @lightsighter do you want the deprecated-ness to show up in a legible way in the docs?  (I think that means manually putting `\deprecated` tags in the doxygen comments.)

**Comment by lightsighter** [comment]
Yeah, we can assign this issue to me and I will go through and do it at some point.

**Comment by lightsighter** [comment]
This is fixed with 2b68918
